safetensors/src/tensor.rs

```diff
@@ -388,26 +400,26 @@ impl<'data> SafeTensors<'data> {
     /// The tensor returned is merely a view and the data is not owned by this
     /// structure.
     pub fn tensor(&self, tensor_name: &str) -> Result<TensorView<'data>, SafeTensorError> {
-        if let Some(index) = &self.metadata.index_map.get(tensor_name) {
-            if let Some(info) = &self.metadata.tensors.get(**index) {
-                Ok(TensorView {
-                    dtype: info.dtype,
-                    shape: info.shape.clone(),
-                    data: &self.data[info.data_offsets.0..info.data_offsets.1],
-                })
-            } else {
-                Err(SafeTensorError::TensorNotFound(tensor_name.to_string()))
-            }
-        } else {
-            Err(SafeTensorError::TensorNotFound(tensor_name.to_string()))
-        }
+        let &Some(&index) = &self.metadata.index_map.get(tensor_name) else {
+            return Err(SafeTensorError::TensorNotFound(tensor_name.to_string()));
+        };
+
+        let Some(info) = &self.metadata.tensors.get(index) else {
+            return Err(SafeTensorError::TensorNotFound(tensor_name.to_string()));
+        };
+
+        Ok(TensorView {
+            dtype: info.dtype,
+            shape: info.shape.clone(),
+            data: &self.data[info.data_offsets.0..info.data_offsets.1],
+        })
```

Comment:
This is purely style, right ? I personally find the first version more readable, the control flow here is really confusing with the let else syntax. Preferences...